src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -156,7 +159,8 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     val gasLimit = stx.tx.gasLimit
     val config = EvmConfig.forBlock(blockHeader.number, blockchainConfig)
 
-    val world1 = InMemoryWorldStateProxy(storages, blockchainConfig.accountStartNonce, Some(stateRoot))
+    //FIXME When mining we are setting block tag as -1 but this info won't be pruned
+    val world1 = BlockchainImpl(storages).getWorldStateProxy(-1, blockchainConfig.accountStartNonce, Some(stateRoot))
```

Comment:
As we are only simulating a tx here, maybe it would be better to use the read only `WorldStateProxy` from the blockchain?